@JonTheNiceGuy thanks for the quick response! Yes, this is what I had in mind. CI failure is a whitespace issue now: `ERROR: lib/ansible/modules/net_tools/basics/uri.py:272:1: W293 blank line contains whitespace`

--- a/lib/ansible/modules/net_tools/basics/uri.py
+++ b/lib/ansible/modules/net_tools/basics/uri.py
@@ -269,7 +269,7 @@ EXAMPLES = r'''
   until: _result.status == 200
   retries: 720 # 720 * 5 seconds = 1hour (60*60/5)
   delay: 5 # Every 5 seconds
-  
+
 # There are issues in a supporting Python library that is discussed in
 # https://github.com/ansible/ansible/issues/52705 where a proxy is defined
 # but you want to bypass proxy use on CIDR masks by using no_proxy

--- a/lib/ansible/modules/net_tools/basics/uri.py
+++ b/lib/ansible/modules/net_tools/basics/uri.py
@@ -276,19 +276,19 @@ EXAMPLES = r'''
 - name: Work around a python issue that doesn't support no_proxy envvar
   uri:
     follow_redirects: none
-    validate_certs: False
+    validate_certs: false
     timeout: 5
     url: "http://{{ ip_address }}:{{ port | default(80) }}"
   register: uri_data
-  failed_when: False
-  changed_when: False
+  failed_when: false
+  changed_when: false
   vars:
     ip_address: 192.0.2.1
   environment: |
       {
         {% for no_proxy in (lookup('env', 'no_proxy') | replace(',', '') ).split() %}
-          {% if no_proxy| ipaddr | type_debug != 'NoneType' %}
-            {% if ip_address | ipaddr(no_proxy) | type_debug != 'NoneType' %}
+          {% if no_proxy| ipaddr is not none %}
+            {% if ip_address | ipaddr(no_proxy) is not none %}
               "no_proxy": "{{ ip_address }}"
             {% endif %}
           {% endif %}

--- a/lib/ansible/modules/net_tools/basics/uri.py
+++ b/lib/ansible/modules/net_tools/basics/uri.py
@@ -286,11 +286,15 @@ EXAMPLES = r'''
     ip_address: 192.0.2.1
   environment: |
       {
-        {% for no_proxy in (lookup('env', 'no_proxy') | replace(',', '') ).split() %}
-          {% if no_proxy| ipaddr is not none %}
-            {% if ip_address | ipaddr(no_proxy) is not none %}
-              "no_proxy": "{{ ip_address }}"
-            {% endif %}
+        {% for no_proxy in (lookup('env', 'no_proxy') | regex_replace('\s*,\s*', ' ') ).split() %}
+          {% if no_proxy | regex_search('\/') and no_proxy | ipaddr('net') != '' and no_proxy | ipaddr('net') != false and ip_address | ipaddr(no_proxy) is not none and ip_address | ipaddr(no_proxy) != false %}
+            'no_proxy': '{{ ip_address }}'
+          {% elif no_proxy | regex_search(':') != '' and no_proxy | regex_search(':') != false and no_proxy == ip_address + ':' + (port | default(80)) %}
+            'no_proxy': '{{ ip_address }}:{{ port | default(80) }}'
+          {% elif no_proxy | ipaddr('host') != '' and no_proxy | ipaddr('host') != false and no_proxy == ip_address %}
+            'no_proxy': '{{ ip_address }}'
+          {% elif no_proxy | regex_search('^(\*|)\.') != '' and no_proxy | regex_search('^(\*|)\.') != false and no_proxy | regex_replace('\*', '') in ip_address %}
+            'no_proxy': '{{ ip_address }}'
           {% endif %}
         {% endfor %}
       }

